endpoints/api/test/test_endtoend_auth.py

```diff
@@ -41,11 +40,17 @@ def test_entity_search(auth_engine, requires_email, client):
             assert len(results) == 0
 
             # Try a known prefix.
+            # For federationuser(ldap): avoid doing LDAP lookups for Robot accounts (PROJQUAY-5137) #2505
+            # we need to ensure, the user is federate before we can check it
+            with mock_ldap() as ldap:
+                # Verify that the user is logged in and their username was adjusted.
+                (response, _) = ldap.verify_and_link_user("cool.user", "somepass")
```

Comment:
I think this only applies when auth_engine is `mock_ldap`.
Can only apply this when auth_engine is set to`mock_ldap`. Same thing for the assumption changes below.

Reply:
@kleesc I do not understand what you mean in that regards. If I got it right you mean the change should check if the auth_engine is `mock_ldap` ? 
I doubt as even for the other fixtures listed the federation has to be finished. I have not seen a possibility to authenticate with the full federation being finished/setup.

Reply:
Right, since the `mock_ldap` context wouldn't be relevant to `fake_jwt` or `fake_keystone`?